### ISSUE TITLE
Multibyte language support: oni doesn't display complete charactors

### ISSUE THREAD

**Issue by wbs0829** [extr0py/oni#126]
The Chinese characters are twice as wide as English characters.   In the default setting, the neovim can correctly display the characters in terminals. But oni display only the left half of characters. And the right half is blank.

**Comment by extr0py** [comment]
Thanks for the report, @wbs0829 ! Do you happen to have a sample Unicode file that I could use to reproduce the issue? 

I believe it should be a relatively simple fix, but would like to double-check against terminal neovim and GVim to make sure I've handled it properly. Would be great to have this support in.

**Comment by extr0py** [comment]
I think this repros it:
[multibyte.txt](https://github.com/extr0py/oni/files/675890/multibyte.txt)

Wanted to see if Neovim reserved space for the multibyte characters, and it seems like it does. For this, we might be able to simply adjust the font rendering to expand based on the size of the character. This would involve a little bit of refactoring to the RenderCache

**Comment by wbs0829** [comment]
I'm glad to get your response, @extr0py  ! I try the fixed version. The new version makes great efforts for multibyte support, but there are still something wrong.

At firtst, I open oni. The text `全部` at the bottom right corner doesn't display correctly:
![2016-12-30 09-27-28](https://cloud.githubusercontent.com/assets/1736988/21558347/6859db1a-ce73-11e6-92a6-15d634a831f9.png)
 Moreover, I repeatly test, and then I find that the text in the status line isn't correctly displayed  with a certain probability.

Then I open a Chinese file. It seems OK:
![2016-12-30 09-27-50](https://cloud.githubusercontent.com/assets/1736988/21558350/70412b6c-ce73-11e6-8fc8-7f3176b80ef0.png)

But after I key `Ctrl-E`, the display is chaotic:
![2016-12-30 09-28-09](https://cloud.githubusercontent.com/assets/1736988/21558351/74351c74-ce73-11e6-89ae-7c255d76023a.png)
In line 2, the text should be blank, but there are some afterimages of line 1. The context after the last line is the same. Some Chinese characters are still not display correctly.

After I edit another file, I open the test file again and key `Ctrl-E`. There are some afterimages of the previous file:
![2016-12-30 09-31-14](https://cloud.githubusercontent.com/assets/1736988/21558353/77e02562-ce73-11e6-91b9-d41dfdc7f2bd.png)


For the test, I use the following text:
```markdown
# 威廉·汉纳

威廉·邓比·“比尔”·汉纳是美国动画师、导演、制片人、配音演员和卡通画师，他创作的电影和电视动画角色曾在20世纪大部分时间里娱乐全世界数以亿计的观众。

汉纳于1910年在新墨西哥州寇里县出生，年幼时经常随家人搬家。1919年定居加利福尼亚州康普顿后，汉纳成为鹰级童军。1928年在康普顿高中毕业后，他曾短暂就读康普顿市立学院，后因经济大萧条退学。接下来几个月汉纳到处打零工，于1930年加入哈曼与伊辛动画工作室。20世纪30年代期间，汉纳曾参与制作包括《船长与孩子们》在内的多部动画片，工作能力和知名度都日益增长1937年，正在米高梅工作的汉纳结识约瑟夫·巴伯拉，两人开始长达60余年的合作，很快就因制作《猫和老鼠》和真人与动画角色共同演出的电影打响名气。米高梅解散动画部门后，巴伯拉和汉纳于1957年共同创立汉纳巴伯拉动画，成为业界最成功的电视动画制片商，推出的动画片包括《摩登原始人》、《瑜伽熊秀》、《史酷比》、《无敌神猫》、《蓝精灵》、《哈克狗》和《杰森一家》。1967年，塔夫脱广播公司以1200万美元价格收购汉纳巴伯拉动画，但两名创始人继续担任公司首脑，直至1991年塔夫脱广播又将他们的公司转卖给透纳广播公司时止。1996年，透纳广播公司同华纳兄弟母公司时代华纳合并，巴伯拉和汉纳此后继续在新公司担任顾问。

巴伯拉和汉纳先后七次获奥斯卡金像奖，八次获艾美奖。他们的动画片已成为文化符号，创作的动画角色也出现在包括电影、书籍和玩具在内的多种媒体或商品上。汉纳巴伯拉动画的节目在20世纪60年代达到鼎盛，在全世界拥有超过三亿观众，并且已被翻译成20多种语言。
```

**Comment by extr0py** [comment]
Thanks for the update and the additional info!

**Comment by extr0py** [comment]
Reopened this to track. It looks like on scroll, all the cells are dirtied, which overwrites half of the characters. Will look at a more robust fix that also handles this case.

**Comment by extr0py** [comment]
I just merged in PR #290 that should have fixed a lot of these issues. @lasypig was interested in this too.

Please test it out and let me know if you have any issues. I'll close this for now, but feel free to reopen if there are still issues around multibyte character support. Thanks!
